addons/hr_payroll/models/hr_payslip.py

```diff
@@ -0,0 +1,570 @@
+    state = fields.Selection([
+        ('draft', 'Draft'),
+        ('verify', 'Waiting'),
+        ('done', 'Done'),
+        ('cancel', 'Rejected'),
+    ], 'Status', index=True, readonly=True, copy=False, default='draft',
+        help='* When the payslip is created the status is \'Draft\'.\
+            \n* If the payslip is under verification, the status is \'Waiting\'. \
+            \n* If the payslip is confirmed then status is set to \'Done\'.\
+            \n* When user cancel payslip the status is \'Rejected\'.')
+    line_ids = fields.One2many('hr.payslip.line', 'slip_id', 'Payslip Lines', readonly=True, states={'draft': [('readonly', False)]})
+    company_id = fields.Many2one('res.company', 'Company', default=lambda self: self.env.user.company_id, readonly=True, states={'draft': [('readonly', False)]}, copy=False)
+    worked_days_line_ids = fields.One2many('hr.payslip.worked_days', 'payslip_id', string='Payslip Worked Days', readonly=True, states={'draft': [('readonly', False)]})
+    input_line_ids = fields.One2many('hr.payslip.input', 'payslip_id', 'Payslip Inputs', readonly=True, states={'draft': [('readonly', False)]})
+    paid = fields.Boolean('Made Payment Order ? ', readonly=True, states={'draft': [('readonly', False)]}, copy=False)
+    note = fields.Text('Internal Note', readonly=True, states={'draft': [('readonly', False)]})
+    contract_id = fields.Many2one('hr.contract', 'Contract', readonly=True, states={'draft': [('readonly', False)]})
+    details_by_salary_rule_category = fields.One2many('hr.payslip.line', compute='_compute_details_by_salary_rule_category', string='Details by Salary Rule Category')
+    credit_note = fields.Boolean(readonly=True, default=False, states={'draft': [('readonly', False)]}, help="Indicates this payslip has a refund of another")
+    payslip_run_id = fields.Many2one('hr.payslip.run', 'Payslip Batches', readonly=True, states={'draft': [('readonly', False)]}, copy=False)
+    payslip_count = fields.Integer(compute='_compute_payslip_count', string='Payslip Computation Details')
+
+    @api.one
+    @api.constrains('date_from', 'date_to')
+    def check_dates(self):
+        if not self._check_dates():
+            raise ValidationError(
+                _("Payslip 'Date From create must be before 'Date To'"))
+
+    @api.multi
+    def _compute_payslip_count(self):
+        payslip_data = self.env['hr.payslip.line'].read_group([('slip_id', 'in', self.ids)], ['slip_id'], ['slip_id'])
+        mapped_data = dict([(m['slip_id'][0], m['slip_id_count']) for m in payslip_data])
+        for payslip in self:
+            payslip.payslip_count = mapped_data.get(payslip.id, 0)
+
+    @api.multi
+    def _compute_details_by_salary_rule_category(self):
+        result = {}
+        if not self.ids:
+            return result
+        for id in self.ids:
+            result.setdefault(id, [])
```

Comment:
this function should be improved.. no need to declare empty result dictionary. 
it's compute function we can directly update field value...